### ISSUE TITLE
Does not build when running react-native run-windows

### ISSUE THREAD

**Issue by jiankai82** [rebeccahughes/react-native-device-info#151]
If you add and link this plugin to your Windows UWP react native app and you run react-native run-windows to build the app, you will get an error saying failed to restore Nuget Package.  This is because the project reference to React Native Windows project has incorrect path.  In react-native-device-info/windows/RNDeviceInfo/RNDeviceInfo.csproj, this line is incorrect: `<ProjectReference Include="..\..\node_modules\react-native-windows\ReactWindows\ReactNative\ReactNative.csproj">`

The correct path should be: `<ProjectReference Include="..\..\..\react-native-windows\ReactWindows\ReactNative\ReactNative.csproj">`

This is not a problem if you build your app by opening your app's solution file in Visual Studio and build using Visual Studio.

Can someone please fix this ASAP?  Thanks.

**Comment by kevinvangelder** [comment]
@jiankai82 Did you by chance try to put screenshots? If so, I can't see them. Can you type out the two paths or try to fix the images?

**Comment by jiankai82** [comment]
Hmm....  I did type the paths in my original post.  I think GitHub might have removed it for security purpose.  I'll try again.

Open react-native-device-info/windows/RNDeviceInfo/RNDeviceInfo.csproj
locate line 116
The Include attribute has the incorrect path which is 
..\..\node_modules\react-native-windows\ReactWindows\ReactNative\ReactNative.csproj
It should be:
..\..\..\react-native-windows\ReactWindows\ReactNative\ReactNative.csproj

Please fix it ASAP.  Thanks.

**Comment by jonathantorley** [comment]
Visual Studio automatically fixes this when you try to build your application (it probably just matches the guid of the project and fixes the reference) but that's obviously no good for a command-line build.

It doesn't make sense to change the project to refer to the new path, as then the project won't compile on its own.

If there is a way to correct the path live Visual Studio does in a pre-build step (maybe Visual Studio uses an msbuild target to do the correction that we can reuse) that would be great, but I haven't got the time to look into it.

In the meantime, I've used the below powershell script to correct the issue.
```powershell
# Get the absolute path of the file
$path = Join-Path -Path $pwd -ChildPath 'node_modules\react-native-device-info\windows\RNDeviceInfo\RNDeviceInfo.csproj'

# Read the xml document
$xml = [xml](Get-Content $path)

# Namespaces, because we need it even for the default (and it can't be xmlns)
$ns = @{ dns = 'http://schemas.microsoft.com/developer/msbuild/2003' }

# Specify the old value (to match against) and the new value (to replace it with)
$oldReference = '..\..\node_modules\react-native-windows\ReactWindows\ReactNative\ReactNative.csproj'
$newReference = "..\..\..\react-native-windows\ReactWindows\ReactNative\ReactNative.csproj"

# Find the items to replace
$projectReferences = Select-Xml `
  -Xml $xml `
  -XPath "/dns:Project/dns:ItemGroup/dns:ProjectReference[@Include='$($oldReference)']" `
  -Namespace $ns

# Replace them
foreach ($projectReference in $projectReferences) {
  $projectReference.node.Include = $newReference
}

# Save it back into the file
$xml.Save($path)
```

**Comment by machour** [comment]
Anyone's up for a PR to address this?

**Comment by machour** [comment]
Fixed in 0.24.3 by @jmcginty, please open a new issue if you're still having problems.
